File: src/automatica.core/Automatica.WebNew/src/app/base/model/areas/area-template.ts
import { Model, BaseModel, JsonFieldInfo, JsonProperty } from "../base-model";

@Model()
export class AreaTemplate extends BaseModel {


    @JsonProperty()
    ObjId: string;

    @JsonProperty()
    Name: string;

    @JsonProperty()
    Description: string;

    @JsonProperty()
    Icon: string;

    @JsonProperty()
    This2AreaType: string;

    @JsonProperty()
    ProvidesThis2AreayType: string;

    @JsonProperty()
    NeedsThis2AreaType: string;

    @JsonProperty()
    This2AreaTypeNavigation: AreaTemplate;

    @JsonProperty()
    ProvidesThis2AreayTypeNavigation: AreaTemplate;

    @JsonProperty()
    NeedsThis2AreaTypeNavigation: AreaTemplate;

    @JsonProperty()
    IsDeleteable: boolean;

    public typeInfo(): string {
        return "AreaTemplate";
    }

    protected getJsonProperty(): Map<string, JsonFieldInfo> {
        return void 0;
    }

}

File: src/automatica.core/Automatica.Core.EF/Models/Areas/AreaTemplate.cs
using System;
using Automatica.Core.Model;

namespace Automatica.Core.EF.Models.Areas
{
    public class AreaTemplate : TypedObject
    {
        public Guid ObjId { get; set; }

        public string Name { get; set; }
        public string Description { get; set; }

        public Guid This2AreaType { get; set; }

        public Guid ProvidesThis2AreayType { get; set; }
        public Guid NeedsThis2AreaType { get; set; }

        public string Icon { get; set; }

        public bool IsDeleteable { get; set; }


        public AreaType This2AreaTypeNavigation { get; set; }
        public AreaType ProvidesThis2AreayTypeNavigation { get; set; }
        public AreaType NeedsThis2AreaTypeNavigation { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset ModifiedAt { get; set; }
    }
}
